### PR TITLE
bs4 raw application

### PR DESCRIPTION
**Pull request by charlienparker**
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit)
- [x] Tests for the changes have been added (for bugfixes/features), they use let helpers and before blocks
- [x] For all UI changes, there is cucumber coverage
- [x] Any endpoint touched in the PR has an appropriate Pundit policy. For open endpoints, reasoning is documented in PR and code
- [x] Any endpoint modified in the PR only responds to the expected MIME types.
- [x] For all scripts or rake tasks, how to run it is documented on both the PR and in the code
- [x] There are no inline styles added
- [x] There are no inline javascript added
- [x] There is no hard coded text added/updated in helpers/views/Javascript. New/updated translation strings do not include markup/styles, unless there is supporting documentation
- [x] Code does not use .html_safe
- [x] All images added/updated have alt text
- [x] Doesn’t bypass rubocop rules in any way

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update to a version)

# What is the ticket # detailing the issue?

Ticket: https://www.pivotaltracker.com/story/show/187498552

# A brief description of the changes

Current behavior: Raw application uses old UI.

New behavior:  Raw application uses new UI with the flag on.

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name: `BS4_ADMIN_FLOW_IS_ENABLED`

- [ ] DC
- [x] ME

# Additional Context

## Background
This PR updates the Raw Application (`raw_application.html.erb`) page in the admin flow to work under the bs4 admin feature flag, ensuring functionality and 508 compliance. Trivially, it also follows the new mockups.

I took this opportunity to DRY up the review page front end. Previously, we already DRYed up the two consumer review pages (`review.html.erb` & `review_and_submit.html.erb`) during the consumer uplift work by:
1. Introducing the `FinancialAssistance::Helpers::ReviewAndSubmitHelper` module which tied translation keys to field values to be used by both pages
2. Introducing the `review_and_submit_subsection` partial which cut down on repeated markup

I took this a step further, as the admin page touched on this PR shared much of the same front end as at the two consumer pages. The two consumer pages and this admin page have identical UI, save for the following differences in the applicants section: 
1. Admin has a demographics section
2. Different rows are used entirely in the personal info section
3. Consumer versions conditionally display rows in the other five sections.

The legacy version of this page already attempted to DRY up this page locally using two yaml files which held the key value mappings which the view then iterated over, although the two consumer pages completely ignored this approach and defined everything in the markup. This problem still remained with the earlier consumer refactor; while the `ReviewAndSubmitHelper` helped abstract away the key value mappings for consumer, this admin page still relied on its yaml files.

## Changes

To solve this, I've done away with the `ReviewAndSubmitHelper` module and have replaced it with `FinancialAssistance::Services::SummaryService` class. This new helper class exposes a `sections` method which returns all of the sections - Applicants summaries, My Preferences, etc., used on a review page. Its nested `ApplicantSummary` class oversees the applicant summary section of the review page, and this class's two subclasses `ConsumerApplicantSummary` and `AdminApplicantSummary` drive the differences between the consumer and admin versions of the summary. 

It also borrows from the legacy solution by defining a new yaml file which holds all possible rows used in the applicant summary. The admin page unconditionally uses almost all of its data, while the consumer pages ignore the demographic section and apply some conditional logic on the rows in the other sections.

Moreover, I've refactored `review_and_submit_subsection` as `summary_subsection`, with the main difference being that it allows for some recursive calls in cases of nested rows. This partial now also handles all abstracted sections on its own, meaning that the review benefits partials (`review_benefits.html.erb` & `review_esi_benefit.html.erb`) were rolled back to their pre-uplift form. Relatedly, I've introduced a new partial, `summary.html.erb`, which holds all of the shared markup used by the three pages, simply accepting a `SummaryService` instance used to drive the sections.

### CODE REVIEW

File: components/financial_assistance/app/views/financial_assistance/applications/_review_esi_benefit.html.erb [hunk 1]
```diff
@@ -1,94 +1,85 @@
...
+    <% end %>
 
-      <% if benefit.employer_address.address_2.present? %>
-        <div class="row ptb no-buffer row-height">
-          <div class="col-md-6">EMPLOYER ADDRESSS LINE 2</div>
-          <div class="col-md-6"><%= benefit.employer_address.address_2 %></div>
-        </div>
-      <% end %>
+    <div class="row ptb no-buffer row-height">
+      <div class="col-md-6">CITY</div>
```
Comment (TristanB17) [inline components/financial_assistance/app/views/financial_assistance/applications/_review_esi_benefit.html.erb:20]: Looks like there might be a few places in this view that need translations?

File: components/financial_assistance/app/views/financial_assistance/applications/_review_esi_benefit.html.erb [hunk 2]
```diff
@@ -1,94 +1,85 @@
...
+      <div class="col-md-6">EMPLOYER ADDRESSS LINE 1</div>
+      <div class="col-md-6"><%= benefit.employer_address.address_1 %></div>
     </div>
-    <% if !FinancialAssistanceRegistry.feature_enabled?(:disable_employer_address_fields) %>
+
+    <% if benefit.employer_address.address_2.present? %>
       <div class="row ptb no-buffer row-height">
-        <div class="col-md-6">EMPLOYER ADDRESSS LINE 1</div>
-        <div class="col-md-6"><%= benefit.employer_address.address_1 %></div>
+        <div class="col-md-6">EMPLOYER ADDRESSS LINE 2</div>
```
[the remark this answers is not in the document]
Comment (charlienparker) [reply components/financial_assistance/app/views/financial_assistance/applications/_review_esi_benefit.html.erb:14]: this is a legacy code

this partial was originally updated for bs4 some months ago, but it is no longer being used when bs4 is on

File: components/financial_assistance/app/views/financial_assistance/applications/_review_esi_benefit.html.erb (hunk 1 above)
Comment (charlienparker) [reply components/financial_assistance/app/views/financial_assistance/applications/_review_esi_benefit.html.erb:20]: this is a legacy code

this partial was originally updated for bs4 some months ago, but it is no longer being used when bs4 is on

File: components/financial_assistance/app/controllers/financial_assistance/applications_controller.rb [hunk 3]
```diff
@@ -8,7 +8,7 @@ class ApplicationsController < FinancialAssistance::ApplicationController
     before_action :set_family
     before_action :find_application, :except => [:index, :index_with_filter, :new]
     if EnrollRegistry.feature_enabled?(:bs4_consumer_flow)
-      before_action :enable_bs4_layout, only: [:application_year_selection, :application_checklist, :edit, :eligibility_results, :review_and_submit, :review,
+      before_action :enable_bs4_layout, only: [:application_year_selection, :application_checklist, :edit, :eligibility_results, :review_and_submit, :review, :raw_application,
```
Comment (kristinmerbach) [inline components/financial_assistance/app/controllers/financial_assistance/applications_controller.rb:11]: raw_application is part of the admin flow not the consumer

File: components/financial_assistance/app/models/financial_assistance/services/summary_service.rb [hunk 4]
```diff
@@ -0,0 +1,474 @@
+# frozen_string_literal: true
+
```
Comment (kristinmerbach) [inline components/financial_assistance/app/models/financial_assistance/services/summary_service.rb:2]: really appreciate the thorough comments here!

**Review by kristinmerbach — DISMISSED** [review]
Big changes, but looks good, tested locally.

File: components/financial_assistance/app/controllers/financial_assistance/applications_controller.rb (hunk 3 above)
Comment (charlienparker) [reply components/financial_assistance/app/controllers/financial_assistance/applications_controller.rb]: fixed

**Review by TristanB17 — APPROVED** [review]
LGTM - solid refactor!

**Review by kristinmerbach — APPROVED** [review]

**Pull request merged**